activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -1867,17 +1867,23 @@ def validate_order_args(args)
         args.each do |arg|
           next unless arg.is_a?(Hash)
           arg.each do |_key, value|
-            unless VALID_DIRECTIONS.include?(value)
+            if value.is_a?(Hash)
+              validate_order_args([value])
+            elsif VALID_DIRECTIONS.exclude?(value)
               raise ArgumentError,
                 "Direction \"#{value}\" is invalid. Valid directions are: #{VALID_DIRECTIONS.to_a.inspect}"
             end
           end
         end
       end
 
+      def flattened_args(order_args)
+        order_args.flat_map { |e| e.class.in?([Hash, Array]) ? flattened_args(e.to_a) : e }
```

Comment:
We should not use `e.class` here because this doesn't consider subclasses of `Hash`. Can you change back to `is_a?`

Reply:
@rafaelfranca good catch. done.